api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -528,4 +529,25 @@ message JwtAuthentication {
   // <http://www.w3.org/TR/cors/#cross-origin-request-with-preflight>`_ regardless of JWT
   // requirements specified in the rules.
   bool bypass_cors_preflight = 4;
+
+  // A map of unique requirement_names to JwtRequirements.
+  // :ref:`requirement_name <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.PerRouteConfig.requirement_name>`
+  // in `PerRouteConfig` uses this map to specify a JwtRequirement.
+  map<string, JwtRequirement> requirement_map = 5;
```

Comment:
Do you plan to have an option for `rules` above mention the map here?

Reply:
Now rules is a list or {RouteMatch, JwtRequirement}.  We could use `oneof` to change it to {RouteMatch, oneof(requirement_name,  JwtRequriement)},  But I don't see much benefit beside removing JwtRequirement duplication.  JwtRequriement is simple,  some duplication is OK, IMHO.  So no such plan.